src/utils.js

```diff
@@ -450,10 +472,36 @@ export function getHttpTransferOptions(options, directBinaryUploadOptions) {
         return transferOptions;
     });
 
-    return {
+    const transferOptions = {
         uploadFiles: convertedFiles,
         headers: directBinaryUploadOptions.getHeaders(),
         concurrent: directBinaryUploadOptions.isConcurrent(),
-        maxConcurrent: directBinaryUploadOptions.getMaxConcurrent()
+        maxConcurrent: directBinaryUploadOptions.getMaxConcurrent(),
     };
+
+    const proxyOptions = getProxyAgentOptions(directBinaryUploadOptions);
+    if (proxyOptions) {
+        const { protocol = 'http:' } = proxyOptions;
+        transferOptions.requestOptions = {
+            agent: protocol === 'https:' ? new HttpsProxyAgent(proxyOptions) : new HttpProxyAgent(proxyOptions)
+        };
+    }
+
+    return transferOptions;
+}
+
+/**
+ * Validates and retrieves basic authentication information from a set of options. Will throw an error
+ * if only one of username or password is provided.
+ * @param {object} options Options from which "username" and "password" properties will be retrieved.
+ */
+export function getBasicAuth(options) {
```

Comment:
Could you put this in one of the files dedicated to proxy code? Since this is clearly proxy code - or create a `proxy-utils` or `auth-utils` file. Because otherwise this file here will become a pot-pourri of methods...

Reply:
@tmathern good feedback, thank you. I moved the function to `http-utils.js` instead.